crates/milli/src/update/index_documents/transform.rs

```diff
@@ -734,12 +739,22 @@ impl<'a, 'i> Transform<'a, 'i> {
                 }
             }
 
-            if is_primary_key(id) || necessary_faceted_field(id) || reindex_vectors {
+            if is_primary_key(id) || reindex_vectors {
                 operations.insert(id, DelAddOperation::DeletionAndAddition);
                 obkv_writer.insert(id, val)?;
-            } else if let Some(operation) = settings_diff.reindex_searchable_id(id) {
-                operations.insert(id, operation);
-                obkv_writer.insert(id, val)?;
+            } else {
+                let facet_operation = necessary_faceted_field(id);
+                let searchable_operation = settings_diff.reindex_searchable_id(id);
+                let operation = facet_operation
+                    // TODO: replace `zip.map` with `zip_with` once stable
+                    .zip(searchable_operation)
+                    .map(|(op1, op2)| op1.merge(op2))
+                    .or(facet_operation)
+                    .or(searchable_operation);
```

Comment:
style, but I found this a bit hard to read.

Would have preferred an explicit match here:

```rust
let operation = match (facet_operation, searchable_operation) {
    (Some(facet_operation), Some(searchable_operation)) => Some(facet_operation.merge(searchable_operation)),
    (Some(operation), None) | (None, Some(operation)) => Some(operation),
    (None, None) => None
};
```